### DISCUSSION TITLE
Update angular monorepo does not update peer deps?

### DISCUSSION THREAD

**Discussion by sommmen** [renovatebot/renovate#19005] [Request Help]
Hiya,

There is an update to the monorepo angular:

![image](https://user-images.githubusercontent.com/8641495/203025604-f6414106-d0c7-4706-80e8-04a4fc89269f.png)

unfortunately this fails because peer dependencies are not updated:

![image](https://user-images.githubusercontent.com/8641495/203025797-a70902e5-66a3-40e3-8e63-d58cd144de2d.png)

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: [REDACTED]
npm ERR! Found: @angular/compiler-cli@14.2.11
npm ERR! node_modules/@angular/compiler-cli
npm ERR!   dev @angular/compiler-cli@"^14.2.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @angular/compiler-cli@"^15.0.0" from @angular-devkit/build-angular@15.0.0
npm ERR! node_modules/@angular-devkit/build-angular
npm ERR!   dev @angular-devkit/build-angular@"^15.0.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /tmp/renovate/cache/others/npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /tmp/renovate/cache/others/npm/_logs/2022-11-21T10_07_10_335Z-debug-0.log
```

It seems like renovate should've updated the peer dependencies also, the diff is:

![image](https://user-images.githubusercontent.com/8641495/203026368-9c992f80-c0af-4719-be1c-a38893f4dbc1.png)

Is this normal behaviour or am i missing some config settings? 

config:

``` json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ]
}
```

**Comment by rarkins** [comment]
Are the `@angular-devkit/`, `angular-eslint/` and `@angular/` dependencies in your `package.json` all published from the same source directory?

**Reply by sommmen** [reply]
> Are the `@angular-devkit/`, `angular-eslint/` and `@angular/` dependencies in your `package.json` all published from the same source directory?

@angular-devkit/build-angular https://github.com/angular/angular-cli
@angular-eslint/builder https://github.com/angular-eslint/angular-eslint
@angular-eslint/eslint-plugin https://github.com/angular-eslint/angular-eslint
@angular-eslint/eslint-plugin-template https://github.com/angular-eslint/angular-eslint
@angular-eslint/schematics https://github.com/angular-eslint/angular-eslint
@angular-eslint/template-parser https://github.com/angular-eslint/angular-eslint
@angular/cli https://github.com/angular/angular-cli
@angular/compiler-cli https://github.com/angular/angular

So thats why it is not working (angular-cli vs angular).

Weird because it does not produce different PRs for angular-eslint, or tries to update those within this monorepo.

![image](https://user-images.githubusercontent.com/8641495/203033907-b50c571a-2526-43e0-8f64-4b83f6f6a41c.png)

**Reply by rarkins** [reply]
So would this be correct?
- For minor updates, group based on source repo (should have synchronized versions)
- For major updates, group all of `@angular/`, `@angular-estlint/` and `@angular-devkit/` upgrades together
